src/pm/hydra2/mpiexec/mpiexec.c

```diff
@@ -928,14 +949,20 @@ int main(int argc, char **argv)
         }
     }
 
+    HYD_check_timed_out(&mpiexec_params.timeout, &exit_status);
+
     /* Print exitcodes if necessary */
     if (mpiexec_params.print_all_exitcodes) {
         HYD_PRINT(stdout, "Exit codes: ");
     }
-    exit_status = 0;
+
     for (i = 0; i < pg->num_downstream; i++) {
         char *curr_nodename = NULL;
 
+        if (mpiexec_params.timeout.timed_out && !exitcodes) {
```

Comment:
A comment here would be good to indicate we do not need to collect all the exitcodes in a timeout scenario.

Reply:
Done